core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -195,6 +200,69 @@ void* HIPHostPinnedSpace::impl_allocate(
 
   return ptr;
 }
+
+void* HIPManagedSpace::allocate(const size_t arg_alloc_size) const {
+  return allocate("[unlabeled]", arg_alloc_size);
+}
+void* HIPManagedSpace::allocate(const char* arg_label,
+                                const size_t arg_alloc_size,
+                                const size_t arg_logical_size) const {
+  return impl_allocate(arg_label, arg_alloc_size, arg_logical_size);
+}
+void* HIPManagedSpace::impl_allocate(
+    const char* arg_label, const size_t arg_alloc_size,
+    const size_t arg_logical_size,
+    const Kokkos::Tools::SpaceHandle arg_handle) const {
+  void* ptr = nullptr;
+
+  if (arg_alloc_size > 0) {
+    if (Kokkos::Impl::num_hip_um_allocations.load() == 0 and
+        Kokkos::show_warnings()) {
+      if (!Kokkos::Impl::hip_driver_check_page_migration(m_device)) {
+        std::cerr
+            << R"warning(The combination of device and system configuration does not "
+             "support page migration between device and host.\n"
+             "HIPManagedSpace might not work as expected.\n"
+             " Please refer to the Rocm documentation on unified memory\n)warning";
```

Comment:
Your raw string literal looks weird.

Reply:
reformatted in [890562](https://github.com/kokkos/kokkos/pull/5112/commits/890562382fab84502f8f6a0b75cb971f6b09eab9)